### PR TITLE
Replace docker-image with registry-image.

### PR DESCRIPTION
**Pull request by d**
For context, the registry-image resource was introduced more than two
years ago as a drop-in replace for the docker-image resource (especially
when you're only fetching, not building). It's leaner, less
resource-intense, faster, and doesn't rely on spawning a Docker daemon.

See https://github.com/concourse/registry-image-resource for more.

### Note to reviewers
* I need help to regenerate the pipeline
* On merge I need help to re-fly all the pipelines I touched (just 3 of them, I believe)

## Pre-merge TODO
- [x] dry-run setting master pipeline
- [x] dry-run setting coverity pipeline
- [x] dry-run setting PR pipeline

## Post merge TODO
- [ ] set master pipeline
- [ ] set coverity pipeline
- [ ] set PR pipeline

### CODE REVIEW

**Comment by d** [comment]
Help! I can't easily run things written in python2 at this point, when I tried running `gen_pipeline.py`, I got this:

```
Traceback (most recent call last):
  File "concourse/pipelines/gen_pipeline.py", line 35, in <module>
    import yaml
ImportError: No module named yaml
```

Should I try to install python 2 (EOL'd) and pip2?

**Comment by d** [comment]
> Help! I can't easily run things written in python2 at this point, when I tried running `gen_pipeline.py`, I got this:
> 
> ```
> Traceback (most recent call last):
>   File "concourse/pipelines/gen_pipeline.py", line 35, in <module>
>     import yaml
> ImportError: No module named yaml
> ```
> 
> Should I try to install python 2 (EOL'd) and pip2?

Aha! I found a hack to run EOL'd software! Pushed a new version that includes the generated pipeline

File: concourse/tasks/aix_remote_cleanup.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@
 platform: linux
 
 image_resource:
-  type: docker-image
```
Comment (chrishajas) [inline concourse/tasks/aix_remote_cleanup.yml:6]: is this file still used anywhere?

File: concourse/tasks/gpdb_github_release.yml [hunk 2]
```diff
@@ -1,18 +0,0 @@
...
-    tag: latest
-
-inputs:
-- name: gpdb_src
-
-outputs:
-- name: release_artifacts
-
-run:
-  path: gpdb_src/concourse/scripts/gpdb_github_release.bash
```
Comment (chrishajas) [inline concourse/tasks/gpdb_github_release.yml:18]: We can delete the bash file too I think (and the other yamls we deleted)

**Review by chrishajas — APPROVED** [review]
LGTM, do we expect the improvements to be noticable?

Also, thanks for the cleanup, I think we can remove the bash files of the deleted yaml files too.

File: concourse/tasks/aix_remote_cleanup.yml (hunk 1 above)
Comment (d) [reply concourse/tasks/aix_remote_cleanup.yml:6]: There was a commit that went along the lines of "we don't have time to maintain this on master, let's remove it from the pipeline" so I was ambivalent whether I should keep it or delete it.

**Comment by d** [comment]
Ha! Positive changes I saw while dry running:

```patch
-     CONFIGURE_FLAGS: --enable-cassert --enable-faultinjector --enable-tap-tests
+     CONFIGURE_FLAGS: --enable-cassert --enable-tap-tests
```

**Comment by d** [comment]
Merged!

Validation / deprecation warnings from setting the coverity / PR pipelines:

### Coverity
```
DEPRECATION WARNING:
  - jobs.coverity_scan.plan[1] : aggregate is deprecated and will be removed in a future version
```

### PR pipeline
```
DEPRECATION WARNING:
  - jobs.compile_and_test_gpdb.plan[0] : aggregate is deprecated and will be removed in a future version
  - jobs.compile_and_test_gpdb.plan[2][2] : aggregate is deprecated and will be removed in a future version
```

**Pull request merged**